keel-api/src/main/kotlin/com/netflix/spinnaker/keel/rest/ApplicationController.kt

```diff
@@ -43,12 +45,22 @@ class ApplicationController(
   ): Map<String, Any> {
     if (includeDetails) {
       val resources = resourceRepository.getSummaryByApplication(application)
+      val hasDeliveryConfig = deliveryConfigRepository.hasDeliveryConfig(application)
+      val constraintStates = if (hasDeliveryConfig) {
+        deliveryConfigRepository.constraintStateFor(application)
+      } else {
+        emptyList()
+      }
 
       return mapOf(
         "hasManagedResources" to resources.isNotEmpty(),
-        "resources" to resources
+        "resources" to resources,
+        "hasDeliveryConfig" to hasDeliveryConfig,
+        "currentEnvironmentConstraints" to constraintStates
       )
     }
-    return mapOf("hasManagedResources" to resourceRepository.hasManagedResources(application))
+    return mapOf(
+      "hasManagedResources" to resourceRepository.hasManagedResources(application),
+      "hasDeliveryConfig" to deliveryConfigRepository.hasDeliveryConfig(application))
```

Comment:
Same as lines 48 and (I'm guessing) 56? Should we move the the corresponding `val` declarations to before the `if` statement to avoid repetition?